ner4opt/ner4opt.py

```diff
@@ -0,0 +1,135 @@
+    entities the current library extracts; CONST_DIR (constraint direction), LIMIT (limit),
+    OBJ_DIR (objective direction), OBJ_NAME (objective name), PARAM (parameter) and VAR (variable).
+
+    This library provides access to four main methods for extracting these entities.
+    A `lexical` model built using lexical features alone, a `lexical_plus` model which in addition to lexical features
+    uses optimization scenario specific features, a `semantic` model which uses a roberta based transformers model and
+    a hybrid model which combines all the best features of both lexical and semantic models to achieve best results on
+    our datasets.
+
+    Attributes
+    -------
+    model_name (str) : Specifies the model to use for extracting the entities.
+                       Options include `lexical`, `lexical_plus`, `semantic` and `hybrid`
+                       Default is `hybrid`
+    use_gpu (bool) :   Specifies the model to use gpu while calculating transformer based features.
+                       Default is False
+
+    Methods
+    -------
+    get_entities()
+
+    Attributes
+    -------
+    text (str) : Sample optimization problem
+
+    Returns named entities of a given optimization problem. A list of dictionaries for each entity
+
+    Keys:
+        Name: start, dtype: int: Starting character index of the entity
+        Name: end, dtype: int: Ending character index of the entity
+        Name: word, dtype: str: Entity phrase
+        Name: entity_group, dtype: str: Type of the entity
+        Name: score, dtype: float: Defines the confidence of the prediction
+    """
+
+    def __init__(self, model_name: str = Constants.HYBRID, use_gpu: bool = False):
+        """Init Ner4Opt class."""
+
+        _root_directory = pathlib.Path(__file__).parent.parent
+
+        self._lexical_model_path = os.path.join(_root_directory, Constants.MODELS_DIRECTORY,
+                                                Constants.LEXICAL_MODEL_NAME)
+        self._lexical_plus_model_path = os.path.join(_root_directory, Constants.MODELS_DIRECTORY,
+                                                     Constants.LEXICAL_PLUS_MODEL_NAME)
+        self._hybrid_model_path = os.path.join(_root_directory, Constants.MODELS_DIRECTORY, Constants.HYBRID_MODEL_NAME)
+
+        self._model_name = model_name
+        self._crf_model = None
+        self._deep_model = None
+
+        # Load model
+        if model_name == Constants.LEXICAL:
+            self._crf_model = joblib.load(self._lexical_model_path)
+
+        elif model_name == Constants.LEXICAL_PLUS:
+            self._crf_model = joblib.load(self._lexical_plus_model_path)
+
+        elif model_name == Constants.SEMANTIC:
+            self._deep_model = load_torch_model(Constants.SEMANTIC_DEEP_MODEL, use_gpu=use_gpu)
+
+        elif model_name == Constants.HYBRID:
+            self._deep_model = load_torch_model(Constants.HYBRID_DEEP_MODEL, use_gpu=use_gpu)
```

Comment:
I am probably missing sth.. what's a hybrid deep model? 

I thought our "HYBRID" is using the SEMANTIC_MODEL